### PR TITLE
test: add NRLMSISE propagation validation tests

### PR DESCRIPTION
**Pull request by kyle-cochran**
2 hour propagation against Orekit with a Spherical/Newtonian gravity model + NRLMSISE atmospheric model results in:
`1.5e-1 position error`
`1e-4 velocity error`

Which is pretty bad 🤔

### CODE REVIEW

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated/Orekit_Drag_NRLMSISE00_450km_2hr_run.csv [hunk 1]
```diff
@@ -0,0 +1,32 @@
+DATE,POS_X,POS_Y,POS_,VEL_X,VEL_Y,VEL_
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated/Orekit_Drag_NRLMSISE00_450km_2hr_run.csv:1]: This looks like 30 seconds of data. Is the filename incorrect or the data?

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 2]
```diff
@@ -1914,6 +1914,94 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, PropAc
...
+)
+{
+    {
+        // Current state and instant setup
+        const Instant startInstant = Instant::DateTime(DateTime::Parse("2023-01-01 00:00:00.000"), Scale::UTC);
+
+        // Reference data setup
+        const Table referenceData = Table::Load(
+            File::Path(Path::Parse("/app/test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated/"
+                                   "Orekit_Drag_base_plus_drag_test.csv")),
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp]: Does this file already exist or was it added in this MR?

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 3]
```diff
@@ -1914,6 +1914,94 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, PropAc
...
+            std::make_shared<CentralBodyGravity>(earthSPtr),
+            std::make_shared<AtmosphericDrag>(earthSPtr, satelliteSystem_),
+        };
+
+        // Setup initial conditions
+        const State state = {
+            startInstant,
+            Position::Meters({referencePositionArrayGCRF[0]}, gcrfSPtr_),
+            Velocity::MetersPerSecond({referenceVelocityArrayGCRF[0]}, gcrfSPtr_),
+        };
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:1969]:
```suggestion
        const State state = {
            startInstant,
            Position::Meters(referencePositionArrayGCRF[0], gcrfSPtr_),
            Velocity::MetersPerSecond(referenceVelocityArrayGCRF[0], gcrfSPtr_),
        };
```
Are the {} braces necessary?

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated/Orekit_Drag_NRLMSISE00_450km_2hr_run.csv (hunk 1 above)
Comment (kyle-cochran) [reply test/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated/Orekit_Drag_NRLMSISE00_450km_2hr_run.csv:1]: 🤦

**Comment by Derollez** [comment]
@kyle-cochran Can you add the input you used to generate the input from Orekit 🙏?

**Comment by Derollez** [comment]
@kyle-cochran The failing tests on invalid input spans are failing because of the previous tests using the extra data file added in here I think. Checking further

**Comment by Derollez** [comment]
Superseded by https://github.com/open-space-collective/open-space-toolkit-astrodynamics/pull/225.

**Pull request closed without merging**